skycoin/skywire
# Feat/skywire app example

Did you run `make format && make check`?

Fixes #	

 Changes:	
-	

How to test this PR:
1. On machine A build with `make build-examples`
2. On machine A run `./skywire-cli config gen -irtm` to create a `example-server-app` entry in config
3. On machine A start the visor and start the `example-server-app` from UI or CLI
4. On machine B build with `make build-examples`
5. Start the visor
6. Register the external app `example-client-app` by `./skywire-cli visor app register -a example-client-app`
7. Copy the proc key printed by the above cli and use it in `./app/example-client-app -procKey <proc-key> -addr <visor-a-pk:45>`
8. Check the logs of the binary for a `hi` reply message
9. Deregister the proc after it's use with `./skywire-cli visor app deregister -k <proc-key>`

`Skyproxy`
1. Start a `visorA` from this branch on remote
2. Start the example http app with `go run ./example/http-server/server.go` this will start the app on port `9080`
3. Start a `visorB` from this branch on local
5. Run `./skywire-cli skyproxy connect <remote-pk> --remoteport 9080 --localport 9000`
6. Open `localhost:9000` on local and access the example http server running on remote
7. List the open proxies with `./skywire-cli skyproxy ls`
8. To stop the proxy use `./skywire-cli skyproxy disconnect <id>` (get the id from  the above step)
